In [27]:
## Import Python libraries and functions
import pandas as pd

### IMPORT DATA SETS

In [28]:
# 20 features used in UW project
Data_UW_20 = pd.read_csv('Data_UW_20.csv')

# 218 features from NHANES dataset
Data_NH_200 = pd.read_csv('Data_NHANES_200.csv')

### VIEW ORIGINAL DATA SET FEATURES

In [29]:
# The UW data set features include data categories of: demographics, patient self-reporting questionnaire,
# physical examination by a medical professional, and laboratory bloodwork results. 

Data_UW_20.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47279 entries, 0 to 47278
Data columns (total 27 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   Demographics_Age                              47279 non-null  float64
 1   Laboratory_BloodLeadSI                        47279 non-null  float64
 2   Laboratory_RedCellDistributionWidth           47279 non-null  float64
 3   Questionnaire_GeneralHealth                   47279 non-null  int64  
 4   Laboratory_UrineAlbumin                       47279 non-null  float64
 5   Demographics_IncomeRatio                      47279 non-null  float64
 6   Examination_ArmCircum                         47279 non-null  float64
 7   Laboratory_BloodCadmium                       47279 non-null  float64
 8   Laboratory_AlbuminSI                          47279 non-null  float64
 9   Questionnaire_SpecialHealthCareEquipment_2.0  47279 non-null 

### SELECT NEW DATA SET FEATURES

In [30]:
# For the new set of features, only laboratory data will be included, so several categories of less
# reliable data (patient self reporting questionnaire, physical examination, etc.) will be deleted.

# Patient age must be kept, because that is the target value, 
# so not all "Demographic" data can be deleted until the age column is renamed. 

# For NHANES data set, drop columns whose name contains 'Dietary' or 'Questionnaire' or 'Examination'
Data_NH_200.drop(list(Data_NH_200.filter(regex='Dietary|Questionnaire|Examination')), axis=1, inplace=True)

In [31]:
Data_NH_200.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47279 entries, 0 to 47278
Data columns (total 73 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   Laboratory_WhiteBloodCellCount              47279 non-null  float64
 1   Laboratory_Cotinine                         47279 non-null  float64
 2   Laboratory_Sodium                           47279 non-null  float64
 3   Laboratory_MeanCellVolume                   47279 non-null  float64
 4   Laboratory_CholesterolSI                    47279 non-null  float64
 5   Laboratory_HepB                             47279 non-null  int64  
 6   Laboratory_Monocyte                         47279 non-null  float64
 7   Laboratory_UricAcid                         47279 non-null  float64
 8   Laboratory_BloodCadmium                     47279 non-null  float64
 9   Laboratory_SegmentedNeutrophils             47279 non-null  float64
 10  Laboratory

In [32]:
# Need to finish removing columns, determine meaning behind year labels, and change name of "age" column.

### Handling Data Sets that are too large

** NOTE ** File Size Problem

I tried to load to GitHub the following two data sets:

    Data_UW_20 (19 features, 47279 observations)
    Data_NH_200 (223 features, 47279 observations)

but both data files were too large for GitHub to accept.

Therefore, I will have to do feature selection and probably some reduction in the number of observations 
in each dataset from my computer. Then, I will upload to GitHub the reduced size data sets:

    Data_UW_20_R --- with reduced number of observations
    Data_NH_20 --- with reduced number of observations, and features reduced from 223 down to 20

When I upload my Jupyter Notebook to GitHub, I will leave the feature selection code and observation reduction code in the Jupyter Notebook, but will have it commented out so that the Jupyter Notebook code can be run. The active Jupyter Notebook code will begin with Baseline Modeling, and running the initial Regression models on the two sets of data. 